### PR TITLE
NETOBSERV-466: Add metric to count packet number in FLP

### PR DESCRIPTION
**Pull request by OlivierCazade**
This add a metric to count packet number.

Story was to add only total packet count but I aligned this with the bytes count, and we can still sum different tag to get the total number of packets.

### CODE REVIEW

**Comment by memodi** [comment]
@OlivierCazade  Also, any reason we're not adding metric for "FlowDirection: 1" ? 

cc @jotak -   I believe this metric would also scale similarly as sent/received_bytes_total

**Comment by jotak** [comment]
@memodi yes ... I'd suggest some improvements in our handling of these metrics. I will create a new JIRA for that.

Basically what I'd like to do is to have some default config to not expose all these metrics. The mechanism to ignore metrics already exist, it works with the `tags` defined [here](https://github.com/netobserv/network-observability-operator/pull/179/files#diff-8d1dc7c5c07a7d81c569256b50addac6916775e82020d97d96bd3c30ee17e22fR8-R12) and the `ignoreMetrics` config [there](https://github.com/netobserv/network-observability-operator/blob/main/config/samples/flows_v1alpha1_flowcollector.yaml#L47).

Basically I think we could only expose `received_bytes_total` by default (btw: I don't like its name ... "received" is confusing, maybe we should use the terms ingress/egress instead) ; other metrics would have to be enabled explicitely.
So we could have this set of tags to control what we want:
- ingress
- egress
- bytes
- packets
- workloads
- namespaces

Also I'd like to introduce new similar metrics that have only namespace labels, not workloads, also disabled by default. So that can be used as a mitigation solution if users see too high cardinality with metrics, putting too much pressure on prometheus & FLP, without entirely turning off telemetry: they could turn off per-workload metrics and turn on per-namespace metrics instead.

As said, I'll create a new JIRA

**Comment by jotak** [comment]
> Also, any reason we're not adding metric for "FlowDirection: 1" ?

+1 I think we should provide that as well (it can catch some traffic not caught on ingress), however, as said above, we may want to disable it through config by default.

**Comment by OlivierCazade** [comment]
Thanks for feedback!

>  Also, any reason we're not adding metric for "FlowDirection: 1" ?

The reason for not catching `FLowDirection: 1` was to be consistent with the `received_bytes_total` metric

> +1 I think we should provide that as well (it can catch some traffic not caught on ingress)

This is true, this mean this metrics are reporting wrong value, should we remove both until we find a way to clean duplicate? May be we have to wait for connection tracking

Would you be fine with splitting these metrics into four ?
Something like  `captured_ingress_bytes_total` `captured_egress_bytes_total` `captured_ingress_packets_total` and `captured_egress_packets_total` (and may be enabling by default only (captured_ingress_bytes_total`).

**Comment by OlivierCazade** [comment]
I modified current metrics tag and default ignored metrics.

This PR also address part of this task :
https://github.com/netobserv/network-observability-operator/pull/179

The only remaining part is about having "light" metrics with less cardinality.

@memodi , the last part is renaiming the metrics. Is it fine or would it create a lot of work on your side?

**Comment by memodi** [comment]
> @memodi , the last part is renaiming the metrics. Is it fine or would it create a lot of work on your side?
if it's absolutely desired, go ahead renaming them. thanks!

File: controllers/flowlogspipeline/metrics_definitions/workload_egress_bytes_total.yaml [hunk 1]
```diff
@@ -6,16 +6,15 @@ details:
...
 tags:
-  - bandwidth
-  - rate
-  - namespace
-  - owner
-  - outgoing
+  - egress
+  - bytes
+  - workloads
+  - namespaces
```
Comment (jotak) [inline controllers/flowlogspipeline/metrics_definitions/workload_egress_bytes_total.yaml:12]: I wouldn't set "namespaces" here, else if we later add namespace-only metrics later, we wouldn't be able to filter them out without impacting this one as well (or we'd need to find a trick to do it)

File: controllers/flowlogspipeline/metrics_definitions/workload_egress_packets_total.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  This metric observes the outgoing traffic
+details:
+  Sum packet number for outgoing traffic per source and destination namespaces and owners
+usage:
+  Evaluate network incoming usage breakdown per source and destination namespaces and owners
+tags:
+  - egress
+  - packets
+  - workloads
+  - namespaces
```
Comment (jotak) [inline controllers/flowlogspipeline/metrics_definitions/workload_egress_packets_total.yaml:12]: same here and same for the other workload metrics

File: controllers/flowlogspipeline/metrics_definitions/workload_egress_bytes_total.yaml (hunk 1 above)
Comment (jotak) [reply controllers/flowlogspipeline/metrics_definitions/workload_egress_bytes_total.yaml:12]: well, depends how we do the namespace-only metrics ... you plan to add label-removal feature like we discussed previously, right? If so, then it's ok for me

File: controllers/flowlogspipeline/metrics_definitions/workload_egress_packets_total.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+encode:
+  type: prom
+  prom:
+    metrics:
+      - name: workload_egress_packets_total
+        type: counter
+        valuekey: Packets
+        filter:
+          key: FlowDirection
+          value: "0"
```
Comment (jotak) [inline controllers/flowlogspipeline/metrics_definitions/workload_egress_packets_total.yaml:22]: should be "1" for egress

File: controllers/flowlogspipeline/metrics_definitions/workload_egress_bytes_total.yaml (hunk 1 above)
Comment (OlivierCazade) [reply controllers/flowlogspipeline/metrics_definitions/workload_egress_bytes_total.yaml:12]: I plan to do the label-removal, but I am not sure that this is the priority.

I removed the namespaces label and I will add it again once the label removal task is done.

**Review by jotak — APPROVED** [review]

**Pull request merged**